### ISSUE TITLE
CI/CD and multiple backend services/databases

### ISSUE THREAD

**Issue by dejanberic** [viceroypenguin/VsaTemplate#63]
Could your approach to database first scaffolding and migration work if multiple web apps are built in a CI/CD pipeline, and published using github actions?
The problem is concurrency where only one app should do the migration while others wait on startup, and migration should not even start until all apps are shutdown, right?
Also, sometimes migration does not involve add/removing columns or tables, but executing sql scripts which adds indexes, or changes foreign keys or some other SQL server specific things.
How would you go about it?

**Comment by viceroypenguin** [comment]
1. I would generally not recommend multiple apps have reference to the same database. If you have multiple web apps, they should be using separate databases and only communicate via api endpoints. This is one of the reasons: with a shared database, you no longer have clear definition of who the owner of the DB is anymore.
  * If you have multiple web apps on multiple databases, then the migration I'm doing here should not have any impact, because it only affects a single database.
3. If you're using horizontal scaling with multiple instance of the same app, I would recommend deploying one instance, and then deploying the rest, to handle this scenario. That said, I've not run into that yet, so I don't have any more specific advice than that. 
4. The migration code I've written here does not care what type of migration script is included, only that it is valid SQL. If you add a script that adds an index, then it will be migrated the same as any other migration script. The rebuild of the ORM tables might be "useless" in that case, but that shouldn't be a big deal.

**Comment by viceroypenguin** [comment]
Closing as no further comment was received. Feel free to reopen or create a new issue if further discussion is desired.
